tools/shared.py

```diff
@@ -1224,6 +1227,11 @@ def __getattr__(self, attr):
         raise AttributeError
 
     def __setattr__(self, attr, value):
+      for legacy_attr, fixed_values, error_message in self.attrs['REMOVED_SETTINGS']:
+        if attr == legacy_attr and value not in fixed_values:
+          logger.error('Invalid command line option -s ' + attr + '=' + str(value) + ': ' + error_message)
+          sys.exit(1)
+
```

Comment:
if the value *is* the proper value, so we do not error here, it might be nice to issue a warning, which would suggest to people to remove it. otherwise this PR would make us no longer give any indication about options we remove (unless they have a different value)?

Reply:
Yes, I think its important that developers get warning around use of legacy options.

In addition, if we do go with such warnings then there is less argument for keeping REMOVED_SETTINGS in the js file, since all users will be made aware of such legacy usage without needing to see the contents of the js file.

Reply:
No, we do not want to issue a warning in this case, because it will hint developers towards a direction that breaks backwards compatibility. There is no benefit in breaking backwards compatibility for the joy of breaking backwards compatibility. We want to guide developers to break backwards compatibility when there is a win to be had from doing so.

Developers who deliberately want to keep supporting multiple Emscripten versions would have the annoyance to have to live with spurious warnings being emitted by the compiler when they are doing the right thing, or build that clumsy network of `emcc -v` version sniffing, to avoid the "to get X you cannot say X" issue.

Reply:
Personally, as a developer, when I update to a new compiler version, I do like to know when the compiler options I'm using are legacy/removed, so that I can cleanup and modernize my build.  Especially, as is the case with `-s` settings, this could mean I can cleanup `ifdef` regions of my code.   But you are right, that only really works if I don't plan to support older versions.